packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -27,6 +27,7 @@ Future<int> runCommandAndStreamOutput(List<String> cmd, {
       // TODO(ianh): This doesn't handle IO buffering (where the data might be split half-way through a line)
       dataLines = dataLines.where((String s) => filter.hasMatch(s)).toList();
     }
+    dataLines = dataLines.where((String s) => s.isNotEmpty).toList();
```

Comment:
Yup, good point. I changed to using a `LineSplitter`, filters, and listeners. Shorter and fewer bugs :)